src/include/function/cast/cast_functions.h

```diff
@@ -87,27 +88,81 @@ inline std::string CastToString::castToStringWithVector(
 
 template<typename SRC, typename DST>
 static inline void numericDownCast(SRC& input, DST& result, const std::string& dstTypeStr) {
-    if (input < std::numeric_limits<DST>::min() || input > std::numeric_limits<DST>::max()) {
+    if ((signed)input < std::numeric_limits<DST>::min() ||
+        (signed)input > std::numeric_limits<DST>::max()) {
+        auto a = std::numeric_limits<DST>::min();
+        auto b = std::numeric_limits<DST>::max();
         throw common::RuntimeException(
             "Cast failed. " + std::to_string(input) + " is not in " + dstTypeStr + " range.");
     }
     result = (DST)input;
 }
 
+template<typename T>
+static inline bool tryCastStringToNum(const char* input, const uint64_t& len, T& result) {
+    // skip leading spaces
+    uint64_t start = 0;
+    while (start < len && isspace(input[start])) {
+        start++;
+    }
+
+    // skip trailing spaces
+    uint64_t end = len - 1;
+    while (end > start && isspace(input[end])) {
+        end--;
+    }
+    end++;
+
+    auto out = std::from_chars(input + start, input + end, result);
```

Comment:
This line doesn't compile on mac with clang++14.
`in instantiation of function template specialization 'kuzu::function::tryCastStringToNum<double>' requested here in instantiation of function template specialization 'kuzu::function::castStringToNum<double>' requested here candidate function has been explicitly deleted candidate template ignored: requirement 'is_integral<double>::value' was not satisfied [with _Tp = double] candidate function template not viable: requires 4 arguments, but 3 were provided`